### ISSUE TITLE
progressive images not supported

### ISSUE THREAD

**Issue by slyrus** [sharplispers/cl-jpeg#39]
it would be great if we had support for progressive jpeg images.

**Comment by knobo** [comment]
At least for reading.

**Comment by varjagg** [comment]
I know - it's actually the only item on TODO, the bottom of README.md. It's a fair bit of change though, I'd really like to do it in one sitting over few days but so far never had the time. Hope I can look into that during the summer.

**Comment by knobo** [comment]
I don't know if I'm right, but I suspect there are two ways of doing this. One is to decode progressively while reading. Another way is to read the whole file, then convert it to something that can be decoded as if it was a baseline image. And if I'm right, would it be easier to do it the latter way? Maybe so easy that it could be done in one day, instead of few days?

**Comment by varjagg** [comment]
No, it's pretty much the same, the "convert it" part *is* the progressive decoding.
